src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -39,6 +39,14 @@ import { relativeToAbsolute } from '../../nav_links/to_nav_link';
 export const isModifiedOrPrevented = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
   event.metaKey || event.altKey || event.ctrlKey || event.shiftKey || event.defaultPrevented;
 
+export const isActiveNavLink = (currentId: string | undefined, targetId: string): boolean => {
```

Comment:
I am okay with the fix for now. But we might need to modify this if more plugins are rendered from DataExplorer and in the nav menu. Currently only Discover is the case. Might need to find a more general and expandable solution in the future.

Reply:
Agreed. The current solution is not ideal. I can add a comment to this function. This way, we can easily revisit and adapt it as needed in the future, ensuring a more expandable approach.